core/src/main/java/com/graphhopper/util/details/KVStringDetails.java

```diff
@@ -19,32 +19,39 @@
 
 import com.graphhopper.util.EdgeIteratorState;
 
-import static com.graphhopper.util.Parameters.Details.STREET_NAME;
-
 /**
- * Calculate the speed name segments of a Path
+ * Return a String value from the key-values
  *
  * @author Robin Boldt
  */
-public class StreetNameDetails extends AbstractPathDetailsBuilder {
+public class KVStringDetails extends AbstractPathDetailsBuilder {
 
-    private String curStreetName = null;
+    private String curString = null;
+    private String key;
 
-    public StreetNameDetails() {
-        super(STREET_NAME);
+    public KVStringDetails(String name, String key) {
+        super(name);
+        this.key = key;
     }
 
     @Override
     public boolean isEdgeDifferentToLastEdge(EdgeIteratorState edge) {
-        if (curStreetName == null || !curStreetName.equals(edge.getName())) {
-            curStreetName = edge.getName();
+        if (curString == null) {
+            // TODO it would be a bit more efficient if we fetch the Map only once per edge (we don't if more than one KV path detail is requested)
+            //  but then we would have to couple these path detail somehow. Or at least use the slightly more efficient EdgeKVStorage.get method instead of getAll
```

Comment:
But we do not use getAll here?

Reply:
Ah I see. So if this is important maybe there should be `edge.getKeyValue(key)` instead of just `edge.getKeyValues().get(key)`?